collection-scripts/gather_priority_and_fairness

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+BASE_COLLECTION_PATH="/must-gather"
+APF_PATH="${BASE_COLLECTION_PATH}/api_priority_and_fairness"
+
+mkdir -p ${APF_PATH}
+oc get --raw /debug/api_priority_and_fairness/dump_priority_levels > ${APF_PATH}/dump_priority_levels
```

Comment:
will this collect data from all API servers or just one?

Reply:
based on must-gather from https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_must-gather/204/pull-ci-openshift-must-gather-master-e2e-aws/1350052523385294848/artifacts/e2e-aws/gather-must-gather/ it looks like it is just for a single API.

Reply:
It would be nice if we could get data from all API servers.

Would it be easy to place the files in `namespaces/openshift-kube-apiserver/pods/kube-apiserver/kube-apiserver/api_priority_and_fairness` ? (for each instance)

Reply:
Good morning p0lyn0mial! I will look at what is being done for other scripts and if it is possible and update the patch. Thank you for your review!

Reply:
@p0lyn0mial What do you think about it now? It should collect all API Servers, I am filtering on role=master, then scraping the endpoints from all of them in separated directories under` namespaces/openshift-kube-apiserver/pods/kube-apiserver/kube-apiserver/api_priority_and_fairness`

It will create corresponding directories based on API Server targeted.

```
master-1.rcernin46.lab.upshift.rdu2.redhat.com
master-0.rcernin46.lab.upshift.rdu2.redhat.com
master-2.rcernin46.lab.upshift.rdu2.redhat.com

```
And each of those contains:

```
ls -la master-1.rcernin46.lab.upshift.rdu2.redhat.com
total 72
drwxr-xr-x  5 rrobin  staff   160 16 Jan 15:14 .
drwxr-xr-x  6 rrobin  staff   192 16 Jan 15:14 ..
-rw-r--r--  1 rrobin  staff  9796 16 Jan 15:14 dump_priority_levels
-rw-r--r--  1 rrobin  staff  9796 16 Jan 15:14 dump_queues
-rw-r--r--  1 rrobin  staff  9796 16 Jan 15:14 dump_requests
```

Reply:
Unsure why the Tests failed but here it is how it looks now: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_must-gather/204/pull-ci-openshift-must-gather-master-e2e-aws/1350313571078115328/artifacts/e2e-aws/gather-must-gather/


```
pwd
/must-gather-2/registry-build02-ci-openshift-org-ci-op-q54bk8m9-stable-sha256-9d47fc7795d6366555a7b0c2f7d665879bb86e0360b11ac5c321b0999e4085ec/namespaces/openshift-kube-apiserver/pods/kube-apiserver/kube-apiserver

kube-apiserver ls
api_priority_and_fairness

ls -la **/*

api_priority_and_fairness:
total 0
drwxr-xr-x@ 5 rrobin  staff  160 16 Jan 16:50 .
drwxr-xr-x@ 4 rrobin  staff  128 16 Jan 17:09 ..
drwxr-xr-x@ 5 rrobin  staff  160 16 Jan 16:50 ip-10-0-128-30.us-west-2.compute.internal
drwxr-xr-x@ 5 rrobin  staff  160 16 Jan 16:50 ip-10-0-187-34.us-west-2.compute.internal
drwxr-xr-x@ 5 rrobin  staff  160 16 Jan 16:50 ip-10-0-201-215.us-west-2.compute.internal

api_priority_and_fairness/ip-10-0-128-30.us-west-2.compute.internal:
total 72
drwxr-xr-x@ 5 rrobin  staff    160 16 Jan 16:50 .
drwxr-xr-x@ 5 rrobin  staff    160 16 Jan 16:50 ..
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_priority_levels
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_queues
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_requests

api_priority_and_fairness/ip-10-0-187-34.us-west-2.compute.internal:
total 72
drwxr-xr-x@ 5 rrobin  staff    160 16 Jan 16:50 .
drwxr-xr-x@ 5 rrobin  staff    160 16 Jan 16:50 ..
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_priority_levels
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_queues
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_requests

api_priority_and_fairness/ip-10-0-201-215.us-west-2.compute.internal:
total 72
drwxr-xr-x@ 5 rrobin  staff    160 16 Jan 16:50 .
drwxr-xr-x@ 5 rrobin  staff    160 16 Jan 16:50 ..
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_priority_levels
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_queues
-rw-r--r--@ 1 rrobin  staff  11399 16 Jan 16:50 dump_requests
```

Reply:
Hm it actually failed to get those /debug endpoints, I thought they were exposed. I ll check why it didn't work.

Reply:
Check now, I think it should be all good ? https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_must-gather/204/pull-ci-openshift-must-gather-master-e2e-aws/1350343532023386112/artifacts/e2e-aws/gather-must-gather/

Reply:
> `namespaces/openshift-kube-apiserver/pods/kube-apiserver/kube-apiserver/api_priority_and_fairness`

I would prefer under `namespaces/openshift-kube-apiserver/pods/kube-apiserver-ip-10-0-170-247.us-east-2.compute.internal/kube-apiserver/kube-apiserver` or `namespaces/openshift-kube-apiserver/pods/kube-apiserver-ip-10-0-170-247.us-east-2.compute.internal/kube-apiserver/kube-apiserver/api_priority_and_fairness`